example/spec/loadTest/ingestLoad_queueGranPassThruSF.js

```diff
@@ -0,0 +1,253 @@
+
+const granuleCountThreshold = 0.95;
+
+const totalInputPayloads = [];
+const queueGranulesExecutionArns = [];
+const queueGranulesChildExecutionArns = [];
+const testSuffixes = [];
+const totalGranulesCompleted = [];
+
+let config;
+let beforeAllFailed = false;
+let colorConsoleLog = redConsoleLog();
+
+const batchGranulesProcessing = async (nthWorkflow) => {
+  console.log(yellowConsoleLog(), `\n___ Executing ${nthWorkflow}/${totalWorkflowCount} ${workflowName} workflows ___`);
+
+  const testId = createTimestampedTestId(config.stackName, 'GranuleIngestLoadTest');
+  const testSuffix = `${createTestSuffix(testId)}_${randomString()}`;
+
+  const collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+  const provider = { id: `s3_provider${testSuffix}` };
+
+  // populate collections, providers and test data
+  await Promise.all([
+    addCollections(
+      config.stackName,
+      config.bucket,
+      collectionsDir,
+      testSuffix
+    ),
+    addProviders(
+      config.stackName,
+      config.bucket,
+      providersDir,
+      config.bucket,
+      testSuffix
+    ),
+  ]);
+  await updateCollection({
+    prefix: config.stackName,
+    collection,
+    updateParams: { duplicateHandling: 'replace' },
+  });
+
+  const inputPayload = await setupGranulesForIngestLoadTest(
+    true,
+    config.bucket,
+    inputPayloadJson,
+    granuleCountPerWorkflow,
+    granuleRegex,
+    testSuffix
+  );
+  testSuffixes.push(testSuffix);
+
+  const workflowExecution = await buildAndExecuteWorkflow(
+    config.stackName,
+    config.bucket,
+    workflowName,
+    collection,
+    provider,
+    inputPayload
+  );
+
+  queueGranulesExecutionArns.push(workflowExecution.executionArn);
+  totalInputPayloads.push(inputPayload);
+
+  if (workflowExecution.status === 'completed') {
+    console.log(greenConsoleLog(), `\n___ Execution of Workflow ${nthWorkflow} completed with success status ___`);
+  } else {
+    console.log(redConsoleLog(), `\n___ Execution of Workflow ${nthWorkflow} Failed ___\n ${JSON.stringify(workflowExecution)}\n`);
+  }
+
+  const lambdaOutput = await new LambdaStep().getStepOutput(
+    workflowExecution.executionArn,
+    'QueueGranules'
+  );
+  queueGranulesChildExecutionArns.push(lambdaOutput.payload);
+
+  if (lambdaOutput.payload.running.length === granuleCountPerWorkflow) {
+    colorConsoleLog = greenConsoleLog();
+  } else {
+    colorConsoleLog = redConsoleLog();
+    console.log(colorConsoleLog, '\n___ lambdaOutputPayload ExecutionArns ___');
+    lambdaOutput.payload.running.map((runningArn) =>
+      console.log(`${JSON.stringify(runningArn)}`));
+  }
+  console.log(colorConsoleLog,
+    `\n___ Lambda Output of Workflow ${nthWorkflow} has ${lambdaOutput.payload.running.length}/${granuleCountPerWorkflow} of expected arns ___`);
+
+  const completedGranules = [];
+  const incompletedGranules = [];
+
+  const expectedValues = ['completed'];
+
+  await Promise.all(
```

Comment:
If I'm reading this right I don't think you need the `Promise.all`. You're `awaiting` in `map`, which should make this synchronous. I could be misreading though.

Reply:
It wouldn't work without the Promise.all with my current code layout.